### DISCUSSION TITLE
Prettierd/Prettier not applying Tailwind CSS plugin for Class Sorting

### DISCUSSION THREAD

**Discussion by aharoJ** [nvimtools/none-ls.nvim#107] [General]
### Description

I'm encountering an issue where `null-ls` (using `prettierd` for formatting) does not seem to recognize or apply the Tailwind CSS plugin for class sorting in my project files. Despite having both `prettier` and `prettier-plugin-tailwindcss` installed in my project's `node_modules`, class sorting does not occur upon formatting HTML, JavaScript, or other relevant files that utilize Tailwind CSS classes.

### Detailed Observations

I've noticed a discrepancy in how Prettier behaves when invoked from within Neovim through `null-ls` and when run directly from a terminal:

1. **Neovim (`null-ls` invocation)**: Running the command `:!npx prettier --check %` in Neovim reports that all matched files use Prettier code style, implying no need for reformatting.

    ```sh
    !npx prettier --check %
    >  Messages: :!npx prettier --check src/pages/test.js
    > Checking formatting...
    > All matched files use Prettier code style!
    ```

2. **Terminal (Direct invocation)**: However, running Prettier directly in a terminal with `npx prettier --check file_name.js` indicates that there are code style issues in the same file and suggests running Prettier to fix them.

    ```sh
    npx prettier --check file_name.js
    > Checking formatting...
    > [warn] test.js
    > [warn] Code style issues found in the above file. Run Prettier to fix.
    ```

This difference suggests that when `null-ls` invokes Prettier (or `prettierd`), it might not be using the same context, configuration, or environment as when Prettier is run directly from a terminal.

### Dev
If anyone knows how to set it up properly, please let me know. I've looked over both lsp-config and none-ls **documentation** to try and make it work with no sucesss.

**Comment by mochaaP** [comment]
Maybe your cwd is misconfigured? Check if `:pwd` or `:!pwd` matches your project root.

**Reply by aharoJ** [reply]
Nope, that was the initial test.... can I see your config file? Remember I am talking about formatting the `class sorting` because everything else works perfectly. 

[EXAMPLE](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier) 

```html
<!-- Before -->
<button class="text-white px-4 sm:px-8 py-2 sm:py-3 bg-sky-700 hover:bg-sky-800">...</button>

<!-- After -->
<button class="bg-sky-700 px-4 py-2 text-white hover:bg-sky-800 sm:px-8 sm:py-3">...</button>
```

**Comment by gonviegas** [comment]
I've tried everything to make this work (prettier, prettierd, local/global install, various prettier setup locations/files) and realized that whenever I use a plugin in prettier config file, null-ls times out and can't use prettier at all. I can have the "plugins" entry in prettier config but needs to be empty for it to work. 
Btw, my config was working perfectly with null-ls and prettier v3 and prettier tailwind plugin v0.5.

**Reply by aharoJ** [reply]
[NVIM-CONFIG](https://github.com/aharoJ/nvim-config/blob/main/lua/plugins/none-ls.lua)
Hey @gonviegas... I have everything running perfectly. You can check my config for all lsp, formatting, linting and completion.

My issue is strictly related to `class sorting` because **none-ls** has been wonderful so far.
